src/Marten/Linq/QueryHandlers/CachedQueryHandler.cs

```diff
@@ -26,13 +24,20 @@ public CachedQueryHandler(object model, NpgsqlCommand template, IQueryHandler<T>
             _setters = setters;
         }
 
-        public Type SourceType => _handler.SourceType;
+        public Type SourceType => _handler.SourceType;
+
         public void ConfigureCommand(CommandBuilder builder)
         {
             var sql = _template.CommandText;
             for (var i = 0; i < _setters.Length && i < _template.Parameters.Count; i++)
             {
                 var param = _setters[i].AddParameter(_model, builder);
+
+                if (param.Value is Enum)

```

Comment:
Why is this here? Marten has configurable enum storage of either "as strings" or "as int". This would possibly break here.

Reply:
Yup, it's ugly hack. Without this casting it will throw that it doesn't know the Enum type. I need to revisit that but I believe that here it's only for querying and there would be some automated translation. I need to check it again if there is a chance to do it better.